website/content/docs/intro/use-cases/dynamic-config-vault-dynamic-secrets.mdx

```diff
@@ -0,0 +1,269 @@
+
+$ psql -h <POSTGRESIP> -p <POSTGRES_PORT> -U postgres -c "GRANT SELECT ON ALL TABLES IN SCHEMA public TO \"ro\";"
+```
+
+I already have a Vault cluster up and running, and I’m going to enable and configure
+the database secrets engine, so that Vault can connect to my database and generate
+credentials dynamically.
+
+```shell
+$ vault secrets enable database
+Success! Enabled the database secrets engine at: database/
+
+$ vault write database/config/postgresql \
+  plugin_name=postgresql-database-plugin \
+  connection_url="postgresql://{{username}}:{{password}}@<POSTGRES_ADDR>/postgres?sslmode=disable" \
+  allowed_roles=readonly \
+  username="<POSTGRES_USER>" \
+  password="<POSTGRES_PASSWORD>"
+
+Success! Data written to: database/config/postgresql
+```
+
+In my database secrets engine, I’ve configured a connection as well as a role in
+the secrets engine. With this, authorized entities in Vault will be able to generate
+dynamic credentials in the secrets engine.
+
+```shell
+$ tee readonly.sql <<EOF
+CREATE ROLE "{{name}}" WITH LOGIN PASSWORD '{{password}}' VALID UNTIL '{{expiration}}' INHERIT;
+GRANT ro TO "{{name}}";
+EOF
+
+$ vault write database/roles/readonly \
+  db_name=postgresql \
+  creation_statements=@readonly.sql \
+  default_ttl=1h \
+  max_ttl=24h
+Success! Data written to: database/roles/readonly
+```
+
+The final steps in Vault are to create a Vault policy, which will allow an entity
+to access the database secrets engine and generate a dynamic credential for our
+database, and to create a token with this policy attached to it:
+
+```shell
+$ tee waypoint-policy.vault.hcl <<EOF
+path "database/creds/readonly" {
+  capabilities = [ "read" ]
+}
+EOF
+
+$ vault policy write waypoint waypoint-policy.vault.hcl
+Success! Uploaded policy: waypoint-policy
+
+$ vault token create -policy=waypoint -ttl=24h
+Key                  Value
+---                  -----
+token                <VAULT_TOKEN>
+token_accessor       <VAULT_TOKEN_ACCESSOR>
+token_duration       24h
+token_renewable      true
+token_policies       ["default" "waypoint"]
+identity_policies    []
+policies             ["default" "waypoint"]
+```
+
+Now that we have things configured in Vault, Waypoint needs to be configured to
+connect to Vault. A configuration source for the Vault plugin will be added like so:
+
+```shell
+$ waypoint config source-set -type=vault -config=addr=https://vault.service.consul:8200 -config=token=<VAULT_TOKEN>
+Configuration set for dynamic source "vault"!
+```
+
+We can double-check that the token and address are what we expect as well:
+
+```shell
+$ waypoint config source-get -type=vault
+  KEY  |           VALUE
+--------+----------------------------
+  addr  | https://vault.service.consul:8200
+  token | <VAULT_TOKEN>
+```
+
+At this point, we aren’t actually using the config source yet, but we can get that
+ready by adding it to our waypoint.hcl file, and committing it to the git repository
+for the project.
+
+```hcl
```

Comment:
You might want to copy this out, and run a `waypoint fmt` on it so everything is aligned!

Reply:
I did this many a time in the Google Doc, and lost some formatting when converting to PR 😅 will do!